### PR TITLE
Issue 252

### PR DESCRIPTION
**Pull request by osumgbachiamaka**
*describe your pull request here*

# Checklist

Before submitting a pull request, please check the following:

* [x] All tests pass for you on your machine for all implementations (all implementations must behave identically)
* [x] New tests have been added for any new functionality or to prevent regressions of a bugfix
* [x] Added a short description of the change to `newsfragments/$issue.bugfix` (for fixes) or `.feature` (for new features) or `.doc` (for documentation-only changes)

### CODE REVIEW

**Comment by owlishDeveloper** [comment]
Please make sure all the tests pass!

**Comment by osumgbachiamaka** [comment]
All tests have passed.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 *describe your pull request here*
-
+Fixed Throw a useful error for `$then`
```
Comment (owlishDeveloper) [inline .github/PULL_REQUEST_TEMPLATE.md]: This line got here by mistake, I think? This file is a template

File: .gitignore [hunk 2]
```diff
@@ -47,4 +47,7 @@ dist/
 
 # Go packages
 src/gopkg.in
-src/github.com
\ No newline at end of file
+src/github.com
+
+#python env file
+env
```
Comment (owlishDeveloper) [inline .gitignore]: Nice! I've been thinking about adding this

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,2364 @@
+{
```
Comment (owlishDeveloper) [inline package-lock.json]: Adding `package-lock.json` is a great idea! However, let's not add it in this PR - I'd rather do it in a separate PR (which you are welcome to make! ;) ). I'd keep one issue / fix per PR.

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
   "author": "",
   "license": "MPL-2.0",
   "dependencies": {
+    "json-e": "^3.0.1",
```
Comment (owlishDeveloper) [inline package.json]: I don't think we need this dependency

File: src/index.js [hunk 5]
```diff
@@ -117,7 +117,13 @@ operators.$if = (template, context) => {
     throw new TemplateError('$if can evaluate string expressions only');
   }
   if (isTruthy(interpreter.parse(template['$if'], context))) {
-    return template.hasOwnProperty('then') ? render(template.then, context) : deleteMarker;
+    // return template.hasOwnProperty('then') ? render(template.then, context) : deleteMarker;
```
Comment (owlishDeveloper) [inline src/index.js]: Please don't leave the previous code commented out. It bloats the source code and makes it more difficult to understand. If the code is no longer needed, it's best to just remove it.

File: src/index.js [hunk 6]
```diff
@@ -117,7 +117,13 @@ operators.$if = (template, context) => {
...
+
+    let error = '$if Syntax error: $then: should be spelled then: (no $).'
```
Comment (owlishDeveloper) [inline src/index.js]: We use Error objects (`SyntaxError`, `TemplateError` etc) instead of strings. We then `throw` those errors as opposed to having an `errorMessage` property in `deleteMarker` object. My advice is to use the existing code as your guide

**Comment by osumgbachiamaka** [comment]
TemplateError used in place of strings @owlishDeveloper

**Comment by owlishDeveloper** [comment]
@osumgbachiamaka we still need to remove `package-lock.json` from this PR. We use yarn in this project. Also, please remove the `json-e` dependency (libraries don't use themselves).

Also, can you please add a test case for this?

**Comment by osumgbachiamaka** [comment]
Done
